apps/explorer/lib/explorer/chain.ex

```diff
@@ -4624,6 +4625,18 @@ defmodule Explorer.Chain do
     |> handle_page(paging_options)
   end
 
+  defp handle_random_access_paging_options(query, empty_options, _is_et) when empty_options in [nil, [], %{}],
+    do: limit(query, ^(@default_page_size + 1))
+
+  defp handle_random_access_paging_options(query, paging_options, is_et) do
+    query
+    |> (&if(paging_options |> Map.get(:page_number, 1) |> process_page_number() == 1,
+          do: &1,
+          else: page_transaction(&1, paging_options)
+        )).()
+    |> handle_page(paging_options, is_et)
+  end
```

Comment:
I also noticed this private function is a duplicate of https://github.com/HorizenLabs/eon-explorer/pull/12/files#diff-3fce0f082a5cb927aa738ef8f7d582824df91d08957971d36a0873efacd88816R4619-R4625. Is the thought behind redefining the function to not make changes to the core blockscout code which could be modified if we pull down their latest code?

Reply:
I think youre right to call out the duplication. I added another commit and got rid of it.